client/vscode/src/link-commands/git-helpers.ts

```diff
@@ -41,17 +44,15 @@ export async function repoInfo(filePath: string): Promise<RepositoryInfo | undef
         // Determine repository root directory.
         const fileDirectory = path.dirname(filePath)
         const repoRoot = await gitHelpers.rootDirectory(fileDirectory)
-
         // Determine file path relative to repository root, then replace slashes
         // as \\ does not work in Sourcegraphl links
         const fileRelative = filePath.slice(repoRoot.length + 1).replace(/\\/g, '/')
-
         let { branch, remoteName } = await gitRemoteNameAndBranch(repoRoot, gitHelpers, log)
-
-        branch = getDefaultBranch() || branch
-
         const remoteURL = await gitRemoteUrlWithReplacements(repoRoot, remoteName, gitHelpers, log)
-
+        // check if current branch exists on sourcegraph. If not, open in HEAD instead
+        const validatedBranch = await checkBranch(remoteURL, branch)
+        // Use default branch if set. If not, use validated branch instead
+        branch = getDefaultBranch() || validatedBranch
```

Comment:
```suggestion
        branch = getDefaultBranch() || await checkBranch(remoteURL, branch)
```

We can avoid doing the hard work of validating the branch when there's a default branch through short-circuit evaluation